microservice/security/src/main/java/com/cumulocity/microservice/security/token/CumulocityOAuthConfiguration.java

```diff
@@ -0,0 +1,85 @@
+import org.springframework.beans.factory.ObjectProvider;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.web.servlet.FilterRegistrationBean;
+import org.springframework.context.EnvironmentAware;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.env.Environment;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.authentication.configuration.AuthenticationConfiguration;
+import org.springframework.security.config.annotation.authentication.configuration.EnableGlobalAuthentication;
+import org.springframework.security.web.AuthenticationEntryPoint;
+
+@Configuration(proxyBeanMethods = false)
+public class CumulocityOAuthConfiguration implements EnvironmentAware {
+
+    private Environment environment;
+
+    @Override
+    public void setEnvironment(Environment environment) {
+        this.environment = environment;
+    }
+
+    @Bean
+    public OAuthPostAuthorizationContextProvider oAuthPostAuthorizationContextProvider(
+            ObjectProvider<MicroserviceSubscriptionsService> subscriptionsService) {
+        String applicationName = environment.getProperty("application.name", "");
+        OAuthPostAuthorizationContextProvider provider = new OAuthPostAuthorizationContextProvider(applicationName);
+        subscriptionsService.ifAvailable(provider::setSubscriptionsService);
+        return provider;
+    }
+
+    @Bean
+    public JwtTokenAuthenticationProvider jwtTokenAuthenticationProvider(
+            JwtAuthenticatedTokenCache jwtAuthenticatedTokenCache ) {
+        return new JwtTokenAuthenticationProvider(environment, jwtAuthenticatedTokenCache);
+    }
+
+    @Configuration(proxyBeanMethods = false)
+    @EnableGlobalAuthentication
+    static class AuthenticationConfigurer {
+
+        @Autowired
+        private JwtTokenAuthenticationProvider jwtTokenAuthenticationProvider;
+
+        @Autowired
+        public void configureAuthenticationManager(AuthenticationManagerBuilder auth) {
+            auth.authenticationProvider(jwtTokenAuthenticationProvider);
+        }
+    }
+
+    @Configuration(proxyBeanMethods = false)
+    static class FilterConfigurer {
+
+        @Autowired
+        private AuthenticationConfiguration authenticationConfiguration;
+        @Autowired(required = false)
+        private AuthenticationEntryPoint authenticationEntryPoint;
+        @Autowired
+        private ContextService<UserCredentials> userContextService;
+
+        @Bean
+        public CumulocityOAuthMicroserviceFilter cumulocityOAuthMicroserviceFilter()
+                throws Exception {
+            return new CumulocityOAuthMicroserviceFilter(authenticationConfiguration.getAuthenticationManager(),
```

Comment:
Can we use bean reference `authenticationConfiguration.getAuthenticationManager()` when `proxyBeanMethods = false`?

Reply:
`org.springframework.security.config.annotation.authentication.configuration.AuthenticationConfiguration#getAuthenticationManager()` is _not_ a `@Bean` method. As mentioned in PR description it's the way it's done in Spring under the hood to facilitate global configuration using `AuthenticationManagerBuilder`.